onnxruntime/contrib_ops/cpu/quantization/matmul_nbits_helper.h

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/common/common.h"
+#include "core/providers/common.h"
+#include "core/util/shape_checker.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace matmul_nbits_helper {
+
+template <typename T = Tensor>
+Status CheckInputs(const T* /*activation*/,
+                   const T* quantized_weight,
+                   const T* scales,
+                   const T* zero_points,
+                   const T* group_index,
+                   const T* bias,
+                   int64_t n,
+                   int64_t k,
+                   int64_t block_size,
+                   int64_t bits) {
+  // activation (A)
+  // quantized_weight (B) : (N, k_blocks, blob_size) or null
+  //                        k_blocks = (K + block_size - 1) / block_size
+  //                        blob_size = (block_size * bits + 7) / 8
```

Comment:
Currently it seems I only see the case that `block_size * bits` is a multiple of 8. I wonder whether we have some test to cover those cases when it is not a multiple of 8. Where do you get this equation, can you add the description link?

Reply:
block_size is 16, 32, ..., 128, so block_size * bits will always be multiple of 8.
I will update it to block_size * bits / 8 for simplicity.